membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -21,36 +21,46 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext) exten
   def getAttributes(identityId: String,
                     identityIdToAccountIds: String => Future[String \/ QueryResponse],
                     subscriptionsForAccountId: AccountId => SubPlanReads[AnyPlan] => Future[Disjunction[String, List[Subscription[AnyPlan]]]],
+                    accountSummaryForAccountId: AccountId => Future[\/[String, AccountSummary]],
+                    paymentMethodForPaymentMethodId: PaymentMethodId => Future[\/[String, PaymentMethodResponse]],
                     dynamoAttributeService: AttributeService,
                     forDate: LocalDate = LocalDate.now()): Future[(String, Option[Attributes])] = {
 
+
     def twoWeekExpiry = forDate.toDateTimeAtStartOfDay.plusDays(14)
 
     val attributesFromDynamo: Future[Option[DynamoAttributes]] = dynamoAttributeService.get(identityId) map { attributes =>
       dynamoAttributeService.ttlAgeCheck(attributes, identityId, forDate)
       attributes
     }
 
-    val zuoraAttributesDisjunction: DisjunctionT[Future, String, Option[ZuoraAttributes]] = for {
+    val zuoraAttributesDisjunction: DisjunctionT[Future, String, Future[Option[ZuoraAttributes]]] = for {
       accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId, identityIdToAccountIds), identityId))
       accountIds = queryToAccountIds(accounts)
-      subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](
-        if(accountIds.nonEmpty) withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds, identityId, subscriptionsForAccountId), identityId)
+      accountSummaries <- EitherT[Future, String, List[AccountSummary]](
```

Comment:
There's quite a lot going on inside this for comprehension now... it might be cleaner to pull the accountSummaries and/or subscriptions checks into a separate function?

Reply:
Agreed. I've pulled the empty checks out of the for comprehension.